docs(target): Introduce direct address targets and explain tradeoffs (#2708)
hashicorp/boundary: website/content/docs/oss/installing/connect-to-dev-target.mdx

## Connect to Your First Target

The default target is a `tcp` target with a default port of `22` (which
corresponds to the default SSH port using TCP). This target is set up with

> `127.0.0.1` as its address. When `boundary connect` is executed against this

Reviewer: ```suggestion
`127.0.0.1` as its address. When you execute `boundary connect` against this
```

target, a local authenticated proxy to the address is established on the
target's default port (`127.0.0.1:22`).

```shell-session
$ boundary connect -target-id ttcp_1234567890
```

There is also a secondary `tcp` target with (`ttcp_0987654321`) which uses host
sets instead of an address. The host sets for this target contain the default

> host, which has the address 127.0.0.1. When boundary connect is executed against

Reviewer: ```suggestion
host, which has the address 127.0.0.1. When you execute `boundary connect` against
```

this target, the single available host will be selected and a local
authenticated proxy to the host is also established on the target's default port
(127.0.0.1:22).

```shell-session
boundary connect -target-id ttcp_0987654321
```

For the purpose of this guide, both of these targets accomplish the same goal,

> however they have different purposes. Please see the our

Reviewer: ```suggestion
however they have different purposes. Refer to the 
```
